packages/node_modules/@webex/plugin-meetings/test/unit/spec/integration/index.js

```diff
@@ -0,0 +1,260 @@
+            .then((m) => {
+              alice.meeting = m.meeting;
+              console.log('alice dialed');
+            });
+        })
+        .then(() => testUtils.waitForSpy(bobMeetingAdded, 'meeting:added'))
+        .then((m) => {
+          bob.meeting = m.meeting;
+        })
+        .catch((err) => {
+          console.log('ERROR JOIN ', err);
+          throw err;
+        });
+    });
+
+    it('bob and alice have meeting object', () => {
+      //* * acknoledging makes a difference */
+      bob.meeting.acknowledge('INCOMING');
+      assert.equal(bob.meeting.sipUri, alice.emailAddress);
+      assert.equal(alice.meeting.sipUri, bob.emailAddress);
+      assert.equal(bob.meeting.state, 'IDLE');
+      assert.equal(alice.meeting.state, 'JOINED');
+    }).timeout(20000);
+
+    it('alice leaves the meeting', () => alice.meeting.leave()
+      .then(() => testUtils.waitForStateChange(alice.meeting, 'LEFT')));
+
+    it('alice and bob meeting object deleted', () => Promise.all([
+      testUtils.waitForCallEnded(alice, bob.emailAddress),
+      testUtils.waitForCallEnded(bob, alice.emailAddress)
+    ])
+      .then(() => {
+        assert.equal(alice.spark.meetings.getMeetingByType('sipUri', bob.emailAddress), null);
+        assert.equal(bob.spark.meetings.getMeetingByType('sipUri', alice.emailAddress), null);
+      }));
+  });
+
+  // Alice calls bob and bob rejects it
+  describe('reject Incoming Call', () => {
+    const aliceMeetingAdded = sinon.spy();
+    const bobMeetingAdded = sinon.spy();
+    it('alice dials bob and bob receives meeting added', () => {
+      bob.spark.meetings.on('meeting:added', bobMeetingAdded);
+      alice.spark.meetings.on('meeting:added', aliceMeetingAdded);
+      return alice.spark.meetings.create(bob.emailAddress)
+        .then((m) => {
+          startMeeting(m);
+          alice.meeting = m;
+          return m.join()
+            .then(() => testUtils.waitForSpy(aliceMeetingAdded, 'meeting:added'))
+            .then((m) => {
+              alice.meeting = m.meeting;
+              console.log('alice dialed');
+            });
+        })
+        .then(() => testUtils.waitForSpy(bobMeetingAdded, 'meeting:added'))
+        .then((m) => {
+          bob.meeting = m.meeting;
+        })
+        .catch((err) => {
+          console.log('ERROR JOIN ', err);
+          throw err;
+        });
+    });
+
+    it('bob and alice have meeting object', () => {
+      assert.equal(bob.meeting.sipUri, alice.emailAddress);
+      assert.equal(alice.meeting.sipUri, bob.emailAddress);
+      assert.equal(bob.meeting.state, 'IDLE');
+      assert.equal(alice.meeting.state, 'JOINED');
+    }).timeout(20000);
+
+    it('bob rejects the meeting', () => {
+      bob.meeting.acknowledge('INCOMING');
+      return bob.meeting.decline('BUSY')
+        .then(() => testUtils.waitForStateChange(bob.meeting, 'DECLINED'));
+    });
+
+    it('alice leaves the meeting', () => {
```

Comment:
since all these tests rely on the previous one, I think it's fine to write one huge test that covers all your flows. Separating them out like this means they could never run in parallel and it's also harder to skip certain ones. Just my opinion.

Reply:
Its hard to find where the test failed sometimes . Its easier when we want to skip or perform a particular test.

Will try to merge in the next PR for similar chunk

Reply:
The tests shouldn't rely on each other. If the same things need set up, they can be done in a `beforeEach`.

Reply:
@mattnorris  The the only question in these test file is the It section. Rather then putting all the code into one IT section 
```
 describe('1:1',()=> {
   it('1:1 successfull')  
} 
```

We are just splitting it into 

```
 describe('1:1',()=> {
   it('Alice dialed bob')  
   it('Bob answered') 
   it('1:1 successfull') 
} 
```

1) We can use the describe to skip the tests
2) I dont think we will have issue with test parallel . I think describe can run on parallel but not it

3) It describes the tests similar to how we do it on the sample app or the api